experimental/framework/tree-react-api/src/propNode.ts

```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ErasedType } from "@fluidframework/core-interfaces";
+import type { TreeNode, TreeLeafValue } from "@fluidframework/tree";
+
+/**
+ * A type erased TreeNode for use in react props.
+ * @remarks
+ * Read content from the node using {@link usePropTreeNode} or {@link usePropTreeRecord}.
+ *
+ * In events where tracking dependencies is not required, the node can be unwrapped using {@link unwrapPropTreeNode}.
+ *
+ * To convert a TreeNode to this type use {@link toPropTreeNode} or {@link toPropTreeRecord}.
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface PropTreeNode<T extends TreeNode> extends ErasedType<[T, "PropTreeNode"]> {}
+
+/**
+ * Type TreeNodes in T as {@link PropTreeNode}s.
+ * @remarks
+ * This only handles a few cases (TreeNode, NodeRecord, arrays) and leaves other types as is.
+ * Users which provide other types (e.g. maps) which contain TreeNodes will need to handle wrapping those themselves if the wrapping is desired.
```

Comment:
Why do we only cover a subset of cases here?

Reply:
Covering all cases is impossible (for example we can't fix object which contain nodes and have methods that read from them), and attempting to cover more cases with mapped types has downsides (like breaking nominal typing, erasing protected members, making the generated types really hard to read, causing compile errors due to recursive types, slower builds, more code, etc.)